docs/core/porting/upgrade-assistant-wcf.md

```diff
@@ -0,0 +1,420 @@
+
+```
+[10:26:43 INF] Initializing upgrade step Upgrade app config files
+[10:26:43 INF] Found 0 app settings for upgrade:
+[10:26:43 INF] Found 0 connection strings for upgrade:
+[10:26:43 INF] 0 web page namespace imports need upgraded:
+```
+
+The final step before this project's upgrade is completed, is to update any out-of-date code references. Based on the type of project you're upgrading, a list of known code fixes is displayed for this step. Some of the fixes may not apply to your project.
+
+```
+9. Update source code
+    a. Apply fix for UA0002: Types should be upgraded
+    b. Apply fix for UA0012: 'UnsafeDeserialize()' does not exist
+    c. Apply fix for UA0014: .NET MAUI projects should not reference Xamarin.Forms namespaces
+    d. Apply fix for UA0015: .NET MAUI projects should not reference Xamarin.Essentials namespaces
+```
+
+In this case, none of the suggested fixes apply to the example project, and this step automatically completes immediately after the previous step was completed.
+
+```
+[10:26:44 INF] Initializing upgrade step Update source code
+[10:26:44 INF] Running analyzers on CalculatorService
+[10:26:48 INF] Identified 0 diagnostics in project CalculatorService
+[10:26:51 INF] Initializing upgrade step Move to next project
+```
+
+#### Completing the upgrade
+
+If there are any more projects to migrate, the tool lets you select which project to upgrade next. When there are no more projects to upgrade, the tool brings you to the "Finalize upgrade" step:
+
+```
+1. [Next step] Finalize upgrade
+
+Choose a command:
+   1. Apply next step (Finalize upgrade)
+   2. Skip next step (Finalize upgrade)
+   3. See more step details
+   4. Configure logging
+   5. Exit
+>
+[10:27:15 INF] Applying upgrade step Finalize upgrade
+[10:27:15 INF] Upgrade step Finalize upgrade applied successfully
+```
+
+Ideally, after successfully running the tool, these modified files should look similar. Please check out the updated version of [CalculatorService.csproj](https://github.com/dotnet/docs/tree/main/docs/core/porting/snippets/upgrade-assistant-wcf-framework/CalculatorSample.Upgraded/CalculatorService/CalculatorService.csproj), [app.config](https://github.com/dotnet/docs/tree/main/docs/core/porting/snippets/upgrade-assistant-wcf-framework/CalculatorSample.Upgraded/CalculatorService/App.config), [wcf.config](https://github.com/dotnet/docs/tree/main/docs/core/porting/snippets/upgrade-assistant-wcf-framework/CalculatorSample.Upgraded/CalculatorService/wcf.config), [CalculatorSample/service.cs](https://github.com/dotnet/docs/tree/main/docs/core/porting/snippets/upgrade-assistant-wcf-framework/CalculatorSample.Upgraded/CalculatorService/service.cs).
+
+Notice that if you want to run both CalculatorClient and CalculatorService, you would need to upgrade CalculatorClient with the tool as well. Because CalculatorService targets .Net6.0 after upgrade, it cannot be referenced by CalculatorClient which still targets .NETFramework.
```

Comment:
```suggestion
Notice that if you want to run both CalculatorClient and CalculatorService, you would need to upgrade CalculatorClient with the tool as well. Because CalculatorService targets .Net6.0 after upgrade, it cannot be referenced by CalculatorClient which still targets .NET Framework.
```
What do you mean it can't be referenced by CalculatorClient? Is there a project to project reference or does the upgrade break compatibility with the existing client application.

Reply:
Actually, that's an outdated comment I will remove! Early on, the client project references the server project for the ICalculator ServiceContract. And then after upgrading the CalculatorService, the CalculatorClient has an error because it cannot reference CalculatorService since one is on .NET 6 one is on .NET Framework. But later I modified the client so there isn't a project to project reference and it won't have that error.